mutiny-core/src/lib.rs

```diff
@@ -655,7 +668,28 @@ impl<S: MutinyStorage> MutinyWallet<S> {
 
     /// Get the sorted activity list for lightning payments, channels, and txs.
     pub async fn get_activity(&self) -> Result<Vec<ActivityItem>, MutinyError> {
-        self.node_manager.get_activity().await
+        // Get activities from node manager
+        let mut activities = self.node_manager.get_activity().await?;
+
+        // Get federation activities and merge them with node manager activities
+        let federation_ids = self.list_federations().await?;
+        for fed_id in federation_ids {
+            let federation_activities = self
+                .federations
+                .lock()
+                .await
+                .get(&fed_id)
+                .ok_or(MutinyError::NotFound)?
+                .get_activity()
```

Comment:
this seems inefficient, we get the ids and individually get each client, can we not do this?

```rust 
let federations = self.federations.lock().await;

for (fed_id, federation) in federations {
	let federation_activities= federation.get_activity().await?;
	activities.extend(federation_activities);
}

```